### PR TITLE
[docs] Update README.md for brief descriptions of optimizer and rewriter tools.

### PR DESCRIPTION
**Pull request by shubhambhokare1**
#TODO Add more details about function-based rewriting once tutorial for function-based rewriting is merged

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,6 +142,56 @@ result = Hardmax(v)
...
+
+### Pattern-based rewriting
+
+For this style of rewriting, the user provides a `target_pattern` that is to be replaced, a `replacement_pattern` and a `match_condition` (pattern rewrite will occur only if the match condition is satisfied). A simple example on how to use the pattern-based rewriting tool is as follows:
+
+```python
+from onnxscript.rewriter import pattern
+
+# Create multiple rules
+rule1 = pattern.RewriteRule(
```
Comment (xadupre) [inline README.md:171]: You should include the vode of both rewriting functions.

File: README.md [hunk 2]
```diff
@@ -142,6 +142,56 @@ result = Hardmax(v)
...
+
+```python
+import onnxscript
+
+onnxscript.optimizer.optimize(onnx_model)
+```
+
+For a detailed summary of all the optimizations applied by the optimizer call, refer to the tutorial [Optimizing a Model using the Optimizer](https://onnxscript.ai/tutorial/optimizer/optimize.html)
+
+### ONNX Rewriter
```
Comment (gramalingam) [inline README.md:159]: We should also have a short paragraph on `ONNX IR`

File: README.md [hunk 3]
```diff
@@ -142,6 +142,56 @@ result = Hardmax(v)
 
 More examples can be found in the [docs/examples](docs/examples) directory.
 
+## ONNX Script Tools
```
Comment (gramalingam) [inline README.md:145]: I think updating the very first paragraph would also be helpful. Saying something like "The repo covers ONNXScript + IR/Optimizer/Rewriter", expanded into a couple of sentences

File: README.md [hunk 4]
```diff
@@ -142,6 +152,87 @@ result = Hardmax(v)
...
+* **Low memory footprint:** mmap'ed external tensors; unified interface for
+  ONNX TensorProto, Numpy arrays and PyTorch Tensors etc. No tensor size
+  limitation. Zero copies.
+* **Straightforward access patterns:** Access value information and traverse the
+  graph topology at ease.
+* **Robust mutation:** Create as many iterators as you like on the graph while mutating it.
+* **Speed:** Performant graph manipulation, serialization/deserialization to Protobuf.
+* **Pythonic and familiar APIs:** Classes define Pythonic apis and still map to
+  ONNX protobuf concepts in an intuitive way.
+* **No protobuf dependency:** The IR does not require protobuf once the model is
```
Comment (gramalingam) [inline README.md:172]: nit: maybe this point can be dropped. I am not sure it is relevant to users at this point. For now, protobuf is still the primary serialized format. This point becomes interesting only if we define other alternative serialization formats and support it.

**Review by gramalingam — APPROVED** [review]
LGTM. Just one minor comment that I added.

**Pull request merged**